### PR TITLE
CLN: enforce a minimum GEOS version 3.3.0

### PR DESCRIPTION
**Pull request by mwtoews**
This PR cleans classes and logic from Shapely that supported GEOS versions before 3.3.0.

And yes, I did test this PR locally against GEOS 3.3.0, which passed most tests.

GEOS 3.3.0 was first released in May 2011, with GEOS 3.3.9 released Sept 2013.

See a range of GEOS versions in a few Linux distros here: https://pkgs.org/search/?q=geos

See also #1000 for Shapely-2.0, which will use a higher version.

### CODE REVIEW

File: tests/test_wkt.py [hunk 1]
```diff
@@ -18,12 +18,12 @@ def pipi4():
...
 def test_wkt(pipi):
     """.wkt and wkt.dumps() both do not trim by default."""
-    assert pipi.wkt == "POINT ({0:.16f} {1:.16f})".format(pi, -pi)
+    assert pipi.wkt == "POINT ({0:.15f} {1:.15f})".format(pi, -pi)
 
 
 def test_wkt(pipi4):
     """.wkt and wkt.dumps() both do not trim by default."""
-    assert pipi4.wkt == "POINT ({0:.16f} {1:.16f})".format(pi*4, -pi*4)
+    assert pipi4.wkt == "POINT ({0:.14f} {1:.14f})".format(pi*4, -pi*4)
```
Comment (jorisvandenbossche) [inline tests/test_wkt.py:26]: those changes seem consistent with what I needed to change on the shapely-2.0 branch, so good!

**Review by jorisvandenbossche — COMMENTED** [review]
Skimmed through it, looks good!

**Review by jorisvandenbossche — APPROVED** [review]

File: tests/test_wkt.py (hunk 1 above)
Comment (mwtoews) [reply tests/test_wkt.py:26]: yes, parts of 3024adc06bdd6f110c87c3e08837790033a23eee where cherry-picked

**Review by sgillies — APPROVED** [review]

**Comment by sgillies** [comment]
Thanks @mwtoews !

**Pull request merged**